src/coreclr/gc/gcbridge.cpp

```diff
@@ -186,6 +186,11 @@ struct ColorData
     // Count of colors that list this color in their otherColors
     unsigned incomingColors : INCOMING_COLORS_BITS;
     unsigned visited : 1;
+    // ColorVisibleToClient for a ColorData* can change over the course of bridge processing which
+    // is problematic. We fix this by setting this flag when a color is detected as visible to client.
+    // Once the flag is set, the color is pinned to being visible to client, even though it could lose
+    // some xrefs, making it not satisfy the BridgelessColorIsHeavy condition.
+    unsigned visibleToClient : 1;
```

Comment:
Do we need to tweak `API_INDEX_BITS`/`INCOMING_COLORS_BITS` to fit into the same 32-bit integer?